src/helpers/files.ts

```diff
@@ -1,46 +1,54 @@
 /* eslint-disable max-lines */
-const { cpus } = require('os')
-
-const { yellowBright } = require('chalk')
-const {
-  existsSync,
-  readJson,
-  move,
-  copy,
-  writeJson,
-  readFile,
-  writeFile,
-  ensureDir,
-  readFileSync,
-} = require('fs-extra')
-const globby = require('globby')
-const { outdent } = require('outdent')
-const pLimit = require('p-limit')
-const { join } = require('pathe')
-const slash = require('slash')
-
-const { MINIMUM_REVALIDATE_SECONDS, DIVIDER } = require('../constants')
+import { cpus } from 'os'
+
+import { NetlifyConfig } from '@netlify/build'
+import { yellowBright } from 'chalk'
+import { existsSync, readJson, move, copy, writeJson, readFile, writeFile, ensureDir, readFileSync } from 'fs-extra'
+import globby from 'globby'
+import { PrerenderManifest } from 'next/dist/build'
+import { Redirect as NextRedirect } from 'next/dist/lib/load-custom-routes'
+import { outdent } from 'outdent'
+import pLimit from 'p-limit'
+import { join } from 'pathe'
+import slash from 'slash'
+
+import { MINIMUM_REVALIDATE_SECONDS, DIVIDER } from '../constants'
+
+import { NextConfig } from './config'
+
+interface Redirect extends NextRedirect {
+  regex: string
+  internal?: boolean
+}
+
+type Rewrites =
+  | {
+      fallback?: Array<Redirect>
+      afterFiles?: Array<Redirect>
+      beforeFiles?: Array<Redirect>
+    }
+  | Array<Redirect>
 
 const TEST_ROUTE = /(|\/)\[[^/]+?](\/|\.html|$)/
 
-const isDynamicRoute = (route) => TEST_ROUTE.test(route)
+export const isDynamicRoute = (route) => TEST_ROUTE.test(route)
 
-const stripLocale = (rawPath, locales = []) => {
+export const stripLocale = (rawPath, locales = []) => {
```

Comment:
```suggestion
export const stripLocale = (rawPath: string, locales: Array<string> = []) => {
```
Or locales could be a type